src/posts/xwrf/index.md

```diff
@@ -0,0 +1,162 @@
+                      memory='125G',
+                      processes=11,
+                      interface='ib0',
+                      queue='compute',
+                      walltime='8:00:00',
+                      asynchronous=0)
+n_workers = 2
+cluster.scale(n_workers)
+client = Client(cluster)
+client.wait_for_workers(n_workers=n_workers)
+client
+```
+
+## Load the WRF output
+
+```{code-cell} ipython3
+import intake
+cat = intake.open_catalog("https://raw.githubusercontent.com/xarray-contrib/xwrf-data/main/catalogs/catalog.yml")
+ssp5_ds = cat["xwrf-sample-ssp585"].to_dask()
+ssp5_ds
+```
+
+As one can see, this output is not very helpful. Heaps of necessary information like WRF model grid coordinates or coordinate index assignments are missing. But we can use `xWRF` to very efficiently fix this.
+
+```{code-cell} ipython3
+%time
+ssp5_ds = ssp5_ds.xwrf.postprocess()
+ssp5_ds
+```
+
+As you can see, the `xWRF` postprocessing is done in no time at all thanks to it leveraging the `dask`'s integration and it's delayed computation feature. This is despite the uncompressed dataset being close to 60GB in size.
+
+```{code-cell} ipython3
+ssp5_ds.nbytes*1E-9
+```
+
+Next to some cleanup of the coordinates, the `xWRF` postprocessing includes the calculation of some basic diagnostics not included in WRF output, namely `air_pressure`, `air_potential_temperature`, `geopotential` and `geopotential_height`.
+
+Using `dask`, the computations for determining these diagnostics are delayed. They will be executed upon using `.compute()`, `.persist()` or `.values` on the variable itself or a variable depending on it.
+
+```{code-cell} ipython3
+ssp5_ds.air_pressure
+```
+
+## Example analysis
+
++++
+
+Let's just imagine that we would want to investigate e.g. how climate change might affect the jet stream in different SSP scenarios. One comparison we might want to make could be between the SSP 585 and the SSP 245 simulations.
+
+So first, we load the SSP 245 dataset, too.
+
+```{code-cell} ipython3
+ssp2_ds = cat["xwrf-sample-ssp245"].to_dask().xwrf.postprocess()
+ssp2_ds
+```
+
+Now, in order to calculate the wind speeds, we have to destagger the x and y wind components. Finally, we can define the pressure levels we want to interpolate the data to and use `xgcm.Grid.transform` in order to perform the vertical interpolation.
+
+```{code-cell} ipython3
+from metpy.calc import wind_speed
+import xgcm
+
+plevs = np.array([350., 300., 250., 200., 150., 100.]) # in hPa
+wind_speeds_js = []
+
+for ds in [ssp2_ds, ssp5_ds]:
+    for field in ['U', 'V']:
+        ds[field] = ds[field].xwrf.destagger().variable
+    ds = ds.metpy.quantify()
+    _wind_speed = wind_speed(ds.U, ds.V).metpy.dequantify()
+    grid = xgcm.Grid(ds, periodic=False)
+    _wind_speed = grid.transform(_wind_speed, 'Z', plevs, target_data=ds.air_pressure.pint.to('hPa').metpy.dequantify(), method='log').persist()
+    wind_speeds_js.append(_wind_speed)
+```
+
++++ {"tags": []}
```

Comment:
Can this be moved elsewhere? Why here?

Reply:
This is legacy from the `ipynb` export to `myst`. Removed it.